libc/stdio/fmt.c

```diff
@@ -701,22 +704,23 @@ static int __fmt_bround(struct FPBits *b, int prec, int prec1) {
   // always know in which direction we must round because of the current
   // rounding mode (note that if the correct value for inc is `false` then it
   // doesn't need to be set as we have already done so above)
-  // The last one handles rounding to nearest
-  current_rounding_mode = fegetround();
-  if (current_rounding_mode == FE_TOWARDZERO ||
-      (current_rounding_mode == FE_UPWARD && b->sign) ||
-      (current_rounding_mode == FE_DOWNWARD && !b->sign))
+  // They use the FLT_ROUNDS value - i.e. 0 -> toward zero, 1 -> to nearest,
+  // ties to even, 2 -> toward positive infinity, 3 -> towards negative infinity
+  // The last if handles rounding to nearest
+  if (b->fpi.rounding == 0 || (b->fpi.rounding == 2 && b->sign) ||
```

Comment:
Wouldn't it be better to say `FPI_Round_zero` and `FPI_Round_up` here?

Reply:
This commit now uses the `FPI_Round_*` values.